### PR TITLE
chore(e2e): update config files

### PR DESCRIPTION
**Pull request by vchimev**
## PR Checklist

- [x] The PR title follows our guidelines: https://github.com/NativeScript/NativeScript/blob/master/CONTRIBUTING.md#commit-messages.
- [ ] There is an issue for the bug/feature this PR is for. To avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it.
- [x] You have signed the [CLA](http://www.nativescript.org/cla).
- [x] All existing tests are passing: https://github.com/NativeScript/NativeScript/blob/master/DevelopmentWorkflow.md#running-unit-tests.
- [ ] Tests for the changes are included - https://github.com/NativeScript/NativeScript/blob/master/WritingUnitTests.md.

This pull request:
- installs `tns-core-modules` and `tns-platform-declarations` packages from source
- updates `tsconfig.json` files
- updates `webpack.config.json` files
- adds `mochawesome-report` folder to repo's root `.gitignore` file and delete unnecessary `.gitignore` files
- delete unnecessary `README.md` files
- delete unnecessary `nsconfig.json` files

... to `tests` and `e2e` applications.

### CODE REVIEW

**Comment by vchimev** [comment]
test --ignoreAll

File: e2e/file-qualifiers/webpack.config.js [hunk 1]
```diff
@@ -135,7 +134,6 @@ module.exports = env => {
                             return /[\\/]node_modules[\\/]/.test(moduleName) ||
                                 /[\\/]tns-core-modules[\\/]/.test(moduleName) || // <-- Needed for snapshot builds with linked modules!
```
Comment (MartoYankov) [inline e2e/file-qualifiers/webpack.config.js]: Ask @vakrilov should this line stay here, should it be in all dev apps, etc.

File: e2e/modal-navigation/app/home/home-page.ts [hunk 2]
```diff
@@ -27,12 +27,6 @@ export function onModalNoPage(args: EventData) {
         closeCallback: () => console.log("home-page modal frame closed"),
         fullscreen: false
     });
-
-    view.showModal("modal-no-page/modal-no-page", {
```
Comment (MartoYankov) [inline e2e/modal-navigation/app/home/home-page.ts]: Ask @manoldonev was this intentional

**Comment by vchimev** [comment]
test --ignoreAll

**Comment by vchimev** [comment]
--ignore modal_navigation#${sha1} ios12 api23

**Comment by SvetoslavTsenov** [comment]
test --ignore modal_navigation#${sha1} ios12 api23

**Pull request merged**